src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,119 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String myString;
 
     @Override
     public String getString() {
-        return null;
+        // Simple check if the String is null or not
+        if (myString == null){
+            return null;
+        }
+        else {
+            return myString;
+        }
     }
 
     @Override
     public void setString(String string) {
+        myString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        //edge cases if String length equals 0
+        if (myString == null || myString.isEmpty()) {
+            return 0;
+        }
+        //excluding all the white space, then add each word into Array
+        String[] words= myString.split("\\s+");
+
+        //return the Array length, which is also the length of the string now
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // Position start at 1, not like index, so throws than error
+        if (n <= 0) {
+            throw new IllegalArgumentException("'n' must be greater than zero.");
+        }
+
+        // n must be smaller than the length of the string
+        if (n > myString.length()) {
+            throw new IndexOutOfBoundsException("'n' is greater than the string length.");
+        }
+
+        //initial an empty string
+        String result = "";
+        int position = 1;
```

Comment:
This variable doesn't feel necessary. It's used in one place and could be replaced with i+1.